tlatools/org.lamport.tlatools/src/tlc2/tool/ModelChecker.java

```diff
@@ -269,7 +269,13 @@ protected int modelCheckImpl() throws Exception
                 }
 
                 // We get here because the checking has been completed.
-                result = EC.NO_ERROR;
+                
+                // Check the postcondition.
+                result = tool.checkPostCondition();
+                if (result != EC.NO_ERROR) {
+                	return result;
```

Comment:
If the assumptions are violated, the method will also exit by returning the result.